prune/strategies.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prune
+
+import (
+	"context"
+	"sort"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewPruneByCountStrategy returns a StrategyFunc that will return a list of
+// resources to prune based on a maximum count of resources allowed.
+// If the max count of resources is exceeded, the oldest resources are prioritized for pruning
+func NewPruneByCountStrategy(count int) StrategyFunc {
+	return func(ctx context.Context, objs []client.Object) ([]client.Object, error) {
+
+		if len(objs) <= count {
+			return nil, nil
+		}
+
+		// sort objects by creation date
+		sortedObjs := objs
+
+		sort.SliceStable(sortedObjs, func(i, j int) bool {
```

Comment:
Do we need stable sorting?

Reply:
We could probably get away without doing sorting, but sorting based on the date was my first thought that I felt would work well. Thinking on it some more, we may be able to do something like:
```go
var objsRemove []client.Object

for len(objs) - len(objsRemove) > count {
    // Loop through all the objs to find the oldest one.
}
```
However I feel like ^ would be less performant than sorting.

If we are more so concerned about using the native sort library that Go has because it uses reflect and is slower I could work on writing a different sort function that can be used to be a little bit more performant.

Any thoughts?

Reply:
Sorry I was not very clear. I guess that there are good reasons for using reflect in the core go library. My question was not on "sorting" but "stable", i.e. whether we could use [sort.Slice](https://pkg.go.dev/sort#Slice) rather than [sort.SliceStable](https://pkg.go.dev/sort#SliceStable)

Reply:
Ah okay. I didn't think that much about the differences between the two. After taking a look at the differences, i feel like it would be fine to use the `sort.Slice` and save a few CPU cycles like you said. Thanks for bringing that up!

Reply:
I made a change to use `sort.Slice` in commit https://github.com/operator-framework/operator-lib/pull/105/commits/230e74e5b2422fe95863c28117c52bbeef99915c